src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -113,121 +156,299 @@ export const MultiSelectDialog = chakra(
+                          {...(onMixedStateChange !== undefined
+                            ? {
+                                isChecked: isAllChecked(id, item),
+                                isIndeterminate: isIndeterminate(id, item),
+                                onChange: onMixedStateChange,
+                              }
+                            : {
+                                isChecked: isChecked(id, item.id),
+                                onChange: onChange,
+                              })}
                         />
-                      )}
-                    </ListItem>
-                  ))}
-              </UnorderedList>
-              {isOpen && isLargerThanMobile && (
-                <ButtonGroup __css={styles.actionButtons}>
-                  <Button
-                    id={`ms-${id}-clear`}
-                    buttonType="link"
-                    type="button"
-                    onClick={onClear}
-                  >
-                    Clear
-                  </Button>
-                  <Button
-                    id={`ms-${id}-apply`}
-                    buttonType="primary"
-                    type="button"
-                    onClick={() => {
-                      // Close the multiselect on apply.
-                      setIsOpen(false);
-                      // Run the onApply prop function.
-                      onApply();
-                    }}
-                  >
-                    Apply
-                  </Button>
-                </ButtonGroup>
-              )}
+
+                        {item.children.map((childItem) => (
+                          <Checkbox
+                            key={childItem.id}
+                            marginInlineStart="0"
+                            __css={styles.menuChildren}
+                            id={childItem.id}
+                            labelText={childItem.name}
+                            name={childItem.name}
+                            isChecked={isChecked(id, childItem.id)}
+                            onChange={onChange}
+                            display={defaultChildCheckboxes}
+                          />
+                        ))}
+                      </>
+                    ) : (
+                      <Checkbox
+                        id={item.id}
+                        labelText={item.name}
+                        name={item.name}
+                        isChecked={isChecked(id, item.id)}
+                        onChange={onChange}
+                      />
+                    )}
+                  </React.Fragment>
+                ))}
+              </CheckboxGroup>
+              {showViewLabel()}
             </Box>
-          </FocusLock>
+          </>
+        );
+      };
+
+      const showViewLabel = () => {
+        if (itemsList.length > 0) {
+          return (
+            <Text
+              onClick={() => viewAllItems()}
```

Comment:
You should not call `onClick` functions on divs or p elements. The `Text` component renders a `p` element. Instead, this should be a button.